### PR TITLE
Attempt to stop card testing activity by using the Google reCAPTCHA SDK

### PR DESCRIPTION
**Pull request by JoemG**
This change implements the Google reCAPTCHA v2 and v3 SDKs behind 2 independent feature switches.

v2 error

**v3 - stripe disabled**
image
**v2 - robot checkbox**
image

### CODE REVIEW

File: support-frontend/app/services/RecaptchaService.scala [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+
+  def verify(token: String, secretKey: String): EitherT[Future, String, RecaptchaResponse] =
+    wsClient.url(recaptchaEndpoint)
+      .withQueryStringParameters("secret" -> secretKey, "response" -> token)
+      .withHttpHeaders("Content-length" -> 0.toString)
+      .withMethod("POST")
+      .execute()
+      .attemptT
+      .leftMap(_.toString)
+      .subflatMap(resp => (resp.json).validate[RecaptchaResponse].asEither.leftMap(_.mkString(",")))
```
Comment (paulbrown1982) [inline support-frontend/app/services/RecaptchaService.scala]: You might have to talk me through the various monad transformations here someday!

File: support-frontend/app/wiring/Controllers.scala [hunk 2]
```diff
@@ -51,6 +53,14 @@ trait Controllers {
     fontLoader
   )
 
+  lazy val recaptchaController = new RecaptchaController(
+    components = controllerComponents,
+    actionRefiners = actionRefiners,
+    recaptchaService = recaptchaService,
+    v3RecaptchaKey = appConfig.recaptchaConfigProvider.v3SecretKey,
+    v2RecaptchaKey = appConfig.recaptchaConfigProvider.v2SecretKey,
```
Comment (paulbrown1982) [inline support-frontend/app/wiring/Controllers.scala]: Might this dangling comma be why the build is failing?

File: support-frontend/app/services/RecaptchaService.scala (hunk 1 above)
Comment (tomrf1) [reply support-frontend/app/services/RecaptchaService.scala]: `attemptT` lifts the `Future` into an `EitherT[Future,Throwable,A]` (which represents a`Future[Either[...]]`).
`subflatMap` flattens the `Either`s

**Comment by paulbrown1982** [comment]
For the record (in Github), we've agreed with stakeholders to go live with a v2 A/B test today.

File: support-frontend/assets/pages/contributions-landing/components/ContributionForm.jsx [hunk 3]
```diff
@@ -256,6 +257,7 @@ function withProps(props: PropTypes) {
           isTestUser={props.isTestUser}
           country={props.country}
         />
+        <div id="robot_checkbox" className={props.paymentMethod === 'Stripe' ? 'robot_checkbox' : 'hidden'} />
```
Comment (tomrf1) [inline support-frontend/assets/pages/contributions-landing/components/ContributionForm.jsx]: To add an error message, we can wrap this in a `div` and display an error if `checkoutFormHasBeenSubmitted && !v2IsLowRisk`

**Pull request merged**